File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/order_trades.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json        = Json(JsonConfiguration.Stable)
// val orderTrades = json.parse(OrderTrades.serializer(), jsonString)

package OrderTrades

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class OrderTrades (
    val data: List<Datum>? = null,
    val status: String? = null
)

@Serializable
data class Datum (
    @SerialName("average_price")
    val averagePrice: Long? = null,

    val exchange: String? = null,

    @SerialName("exchange_order_id")
    val exchangeOrderID: String? = null,

    @SerialName("exchange_timestamp")
    val exchangeTimestamp: String? = null,

    @SerialName("fill_timestamp")
    val fillTimestamp: String? = null,

    @SerialName("instrument_token")
    val instrumentToken: Long? = null,

    @SerialName("order_id")
    val orderID: String? = null,

    @SerialName("order_timestamp")
    val orderTimestamp: String? = null,

    val product: String? = null,
    val quantity: Long? = null,

    @SerialName("trade_id")
    val tradeID: String? = null,

    val tradingsymbol: String? = null,

    @SerialName("transaction_type")
    val transactionType: String? = null
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/order_trades.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderTrades {
    pub data: Option<Vec<Datum>>,
    pub status: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Datum {
    pub average_price: Option<i64>,
    pub exchange: Option<String>,
    pub exchange_order_id: Option<String>,
    pub exchange_timestamp: Option<String>,
    pub fill_timestamp: Option<String>,
    pub instrument_token: Option<i64>,
    pub order_id: Option<String>,
    pub order_timestamp: Option<String>,
    pub product: Option<String>,
    pub quantity: Option<i64>,
    pub trade_id: Option<String>,
    pub tradingsymbol: Option<String>,
    pub transaction_type: Option<String>,
}
